### DISCUSSION TITLE
CoveragePlot

### DISCUSSION THREAD

**Discussion by DLGisch** [stuart-lab/signac#1253] [Q&A] [answered]
Hello,
Dear developers, thank you for your commitment to the development of Signac. I am exploring the tool and want to understand it better. 

I am working with a data set where I could not understand some facts that the CoveragePlot was showing me.
I decided to reproduce the same steps with PBMC data.

Below I plot the PBMC data to expose the points that I don't understand. Here are the 4 CoveragePlots 

A - Original pbmc plot for 7 different idents.
B - Original pbmc plot for 2 idents of the previous group where we see a change in the range, but do not see a change in the peaks.   I think I found the explanation for the change in range through this [post](https://github.com/stuart-lab/signac/issues/992). Is there a way to show only the peaks of the cell types that are being plotted? 
C - I have subset pbmc to two idents and the range has completely changed, but the peaks are still the same.
D - I ran callpeak for PBMC subset and the links changed. In my work data the peaks have also changed. 

![CD8A](https://user-images.githubusercontent.com/23063719/196058809-b10b3715-4517-4102-905a-fecbb0f54c33.png)

I ran the  [vignette](https://stuartlab.org/signac/articles/pbmc_multiomic.html) after Joint UMAP visualization without any modifications than I did the code bellow.
```

```{r}

g <- "GNLY"
gr <- LookupGeneCoords(pbmc, gene = g)
gr@metadata$gene <- g

DefaultAssay(pbmc) <- "peaks"
pbmc <- RegionStats(pbmc, genome = BSgenome.Hsapiens.UCSC.hg38)
# link peaks to genes
pbmc <- LinkPeaks(
  object = pbmc,
  peak.assay = "peaks",
  expression.assay = "SCT",
  genes.use = gr@metadata$gene
)

idents.plot <- c("B naive", "B intermediate", "B memory",
                 "CD14 Mono", "CD16 Mono", "CD8 TEM", "CD8 Naive")
p1 <- CoveragePlot(
  object = pbmc,
  region = gr,
  features = gr@metadata$gene,
  expression.assay = "SCT",
  idents = idents.plot,
  extend.upstream = 8000,
  extend.downstream = 5000,
  #peaks.group.by = c("CD8 Naive")
)

idents.plot <- c("CD14 Mono", "CD16 Mono")
p2 <- CoveragePlot(
  object = pbmc,
  region = gr,
  features = gr@metadata$gene,
  expression.assay = "SCT",
  idents = idents.plot,
  extend.upstream = 8000,
  extend.downstream = 5000
)

pbmc.1 <- subset(pbmc,idents=(c("CD14 Mono", "CD16 Mono")))
DefaultAssay(pbmc.1) <- "peaks"
# first compute the GC content for each peak
pbmc.1 <- RegionStats(pbmc.1, genome = BSgenome.Hsapiens.UCSC.hg38)
# link peaks to genes
pbmc.1 <- LinkPeaks(
  object = pbmc.1,
  peak.assay = "peaks",
  expression.assay = "SCT",
  genes.use = gr@metadata$gene
)

p3 <- CoveragePlot(
  object = pbmc.1,
  region = gr,
  features = gr@metadata$gene,
  expression.assay = "SCT",
  idents = idents.plot,
  extend.upstream = 8000,
  extend.downstream = 5000
)


# get gene annotations for hg38
annotation <- GetGRangesFromEnsDb(ensdb = EnsDb.Hsapiens.v86)
seqlevelsStyle(annotation) <- "UCSC"

pbmc.2<-pbmc.1
# call peaks using MACS2
peaks <- CallPeaks(pbmc.2)

# remove peaks on nonstandard chromosomes and in genomic blacklist regions
peaks <- keepStandardChromosomes(peaks, pruning.mode = "coarse")
peaks <- subsetByOverlaps(x = peaks, ranges = blacklist_hg38_unified, invert = TRUE)

# quantify counts in each peak
macs2_counts <- FeatureMatrix(
  fragments = Fragments(pbmc.2),
  features = peaks,
  cells = colnames(pbmc.2)
)

# create a new assay using the MACS2 peak set and add it to the Seurat object
pbmc.2[["peaks"]] <- CreateChromatinAssay(
  counts = macs2_counts,
  fragments = frags,
  annotation = annotation
)

pbmc.2 <- RegionStats(pbmc2, genome = BSgenome.Hsapiens.UCSC.hg38)

# link peaks to genes
pbmc.2 <- LinkPeaks(
  object = pbmc.2,
  peak.assay = "peaks",
  expression.assay = "SCT",
  genes.use = gr@metadata$gene
)

DefaultAssay(pbmc.2) <- "peaks"

p4 <- CoveragePlot(
  object = pbmc.2,
  region = gr,
  features = gr@metadata$gene,
  expression.assay = "SCT",
  idents = idents.plot,
  extend.upstream = 8000,
  extend.downstream = 5000
)
patchwork::wrap_plots(p1, p2,p3,p4,ncol = 1)

```

Thank you!!! \o/

**Comment by timoast** [comment, marked as answer]
> A - Original pbmc plot for 7 different idents.
> B - Original pbmc plot for 2 idents of the previous group where we see a change in the range, but do not see a change in the peaks. I think I found the explanation for the change in range through this https://github.com/stuart-lab/signac/issues/992. Is there a way to show only the peaks of the cell types that are being plotted?

What do you mean exactly, "show only peaks in the cell types plotted"?

> C - I have subset pbmc to two idents and the range has completely changed, but the peaks are still the same.

The normalized count value is scaled according to the mean accessibility across all cells, so can change if you subset the dataset.

> D - I ran callpeak for PBMC subset and the links changed. In my work data the peaks have also changed.

The peak-gene linkage score is dependent on the peak set, since a background set of peaks is sampled to estimate an empirical null distribution of expected z-scores (see the Signac paper for more information)

**Reply by DLGisch** [reply]
What do you mean exactly, "show only peaks in the cell types plotted"?

Sometimes when I plotted a subset of cells even if the peak does not exist for those cells the track with peaks keeps showing. What I thought  was that when plotting the subset of cells only the peaks referring to those cell types would appear.

Thanks for all the clarifications,
Debora

**Reply by timoast** [reply]
No, that's not possible
